### ISSUE TITLE
Config file support

### ISSUE THREAD

**Issue by ALSchwalm** [ALSchwalm/transient#5]
We should support some kind of config file to avoid monstrously large commands. This would also make it easier for people to document _why_ some QEMU flags needs to be passed.

**Comment by senofsky** [comment]
Any predilections for a specific formatting style? toml? yaml? hopefully not json but i guess it's popular?

**Comment by ALSchwalm** [comment]
I'm not sure (though I'm kind of partial to toml). I think we should define a schema using https://marshmallow.readthedocs.io/en/stable/ so we can deserialize from a bunch of sources and just pass that config object to the `Transient` object.

Also, absolutely not yaml.

**Comment by ALSchwalm** [comment]
I'm thinking toml. With something like:

```
[transient]
ssh-console=true
shared-folder = [
  { host = "/path/on/host", guest = "/path/on/guest" },
  { host = "/path/on/host2", guest = "/path/on/guest2" }
]

[qemu]
no-graphic=true
mem="1G"
```

That said, it is not obvious to me that we can trivially write something to allow arbitrary qemu flags there.

**Comment by senofsky** [comment]
Can you assign this issue to me?
I would use toml for the schema and use https://github.com/uiri/toml for the parser.

Additionally, you mentioned using marshmallow for creating the schema and performing object serialization. However, I'm not entirely sure what the "bunch of sources" that you mentioned are.

Would you kindly delineate what solution you envision?

**Comment by ALSchwalm** [comment]
Ok, so the main change here is that we need to define a config type that is passed to `TransientVm` rather than the current `argparse.Namespace`. Therefore, this config type must be constructible from both the `click` options _and_ whatever config file we might define. But one additional complexity is that we must provide useful errors if we encounter issues while making that config type. In particular, we must be able to report some kind of line number indication if a users passes an invalid option (but one that is still _syntactically_ valid, like a number for the vm name).

I'm not exactly sure how to achieve this. I assume we would use something like marshmellow, but I don't think that's going to give us errors that have any kind of context. I also don't want this to be something that we write a ton of code for. If it works out to be more complex than all of the logic in image.py, for example, then it is probably not worth doing.

So ultimately, I don't have a specific solution envisioned, just a bunch of things I think we need to be able to do. I've assigned you, but if you don't think you're down for this, I definitely understand.

**Comment by ALSchwalm** [comment]
Note that we also need to allow someone to override elements of the provided config, from the command line. So if they do: `transient run -config myconfig.toml -name "foo"` the vm name would be "foo" regardless of the contents of `myconfig.toml`. So I think the basic design here would be something like:

- Deserialize the config if provided
- Otherwise, create an 'empty' default config
- Override the fields of the config with whatever explicit arguments were passed on the cmdline
- Pass the config on to `TransientVm`

There is also the question of how this will work with the non-`run` subcommands. I am inclined to think that you could still pass a config to `delete` and `list` and it would work as expected (by listing or deleting the vm defined by the config).

**Comment by ALSchwalm** [comment]
Fixed in #74
